spec/system/topic_list_focus_spec.rb

```diff
@@ -107,4 +108,29 @@ def focussed_post_id
     expect(page).to have_css("body.navigation-topics")
     expect(focussed_topic_id).to eq(oldest_topic.id)
   end
+
+  shared_examples "navigate to topic" do
+    it "navigates to topic using keyboard" do
+      sign_in(admin)
+      visit("/latest")
+
+      page.send_keys("J")
+
+      expect(page).to have_css(".topic-list-item.selected", count: 1)
+
+      page.send_keys(:enter)
+      expect(page).to have_css("body.archetype-regular")
+      expect(page).to have_css("#topic-title")
+    end
+  end
+
+  context "when experimental_prefetch_allowed_groups is enabled" do
+    before { SiteSetting.experimental_prefetch_allowed_groups = "1" }
+    include_examples "navigate to topic"
+  end
+
+  context "when experimental_prefetch_allowed_groups is disabled" do
+    before { SiteSetting.experimental_prefetch_allowed_groups = "" }
+    include_examples "navigate to topic"
+  end
```

Comment:
I think we need specs for the case where a topic is pre-fetched, and then things like posts/title/likes change before the user navigates to it. If we don't handle that case perfectly, we're gonna end up with a **lot** of weird and hard-to-debug issues.